src/sqlfluff/dialects/dialect_sparksql.py

```diff
@@ -2711,3 +2719,25 @@ class GenerateManifestFileStatementSegment(BaseSegment):
             Ref("TableReferenceSegment"),
         ),
     )
+
+
+class ConvertToDeltaStatementSegment(BaseSegment):
+    """A statement to convert other file formats to Delta.
+
+    https://docs.delta.io/latest/delta-utility.html#convert-a-parquet-table-to-a-delta-table
+    https://docs.databricks.com/delta/delta-utility.html#convert-an-iceberg-table-to-a-delta-table
+    """
+
+    type = "convert_to_delta_statement"
+
+    match_grammar: Matchable = Sequence(
+        "CONVERT",
+        "TO",
+        "DELTA",
+        Ref("FileReferenceSegment"),
+        Sequence("NO", "STATISTICS", optional=True),
+        Ref(
+            "PartitionSpecGrammar",
+            optional=True,
+        ),
```

Comment:
Seems quite long, when this also passes black/flake8:

```suggestion
        Ref("PartitionSpecGrammar", optional=True),
```